modules/services/mbsync.nix

```diff
@@ -54,6 +55,15 @@ in {
       '';
     };
 
+    logDirectory = mkOption {
```

Comment:
Isn't there some standard directory for logs? In XDG it seems [XDG_STATE_HOME](https://specifications.freedesktop.org/basedir-spec/latest/ar01s03.html) is intended for this.

So, unless some other directory is preferred under macos, I would suggest omitting the `logDirectory` option and always use `${config.xdg.stateHome}/<app name>` as the log directory.

Reply:
I *think* using a log directory is an anti-pattern when using `systemd`.

Reply:
@ambroisie Yeah, specifying a log directory for a systemd service would not be sensible. And I'm not very fond of the idea of adding a log directory option, I would prefer simply deciding on a base path where they all go, the user could always override the path by changing the value of XDG_STATE_HOME.

Reply:
Hmmm on second thought, I'm not sure macOS actually uses `XDG_STATE_HOME`. I believe launchd logs internally by default and you use the CLI to access them. There's not a lot of documentation that I can find. I think the two options here are:

1. Remove the output path option entirely, let users do `launchd.agents.mbsync.config.StandardOutPath = myPath` if they so choose.
2. If the user explicitly sets `XDG_STATE_HOME`, then use it. Probably want to use `mkDefault` so they can override it.

I think 2 in practice would work the best, but I'm hesitant to use an XDG variable that technically macOS doesn't actually make use of.

Reply:
CC: @rycee @ambroisie

Reply:
I think that even though it does not use the XDG standard, it does have a standard location which has an analogous usage.

I do not have have any strong feelings regarding its use or not, I was simply stating that (IIRC) using a log directory goes somewhat against what is idiomatic whenever using systemd.

Reply:
Okay so after some further reading, it looks like both systemd and launchd log to some internal mechanism that can be redirected to files using `StandardOutput` and `ErrorOutput` keys in the unit files. So I will just completely remove those keys and leave it up to the user to specify, then we should have basically identical functionality on both the launchd and systemd units.